index.html

```diff
@@ -1051,25 +1057,37 @@ <h3>Data delivery</h3>
               and {{PressureRecord/[[Time]]}} set to |timestamp|.
             </li>
             <li>
-              If |observer|.{{PressureObserver/[[AfterPenaltyRecordMap]]}}[source] exists,
-              set observer.{{PressureObserver/[[AfterPenaltyRecordMap]]}}[source] to |record|
-              and [=iteration/continue=].
+              If |observer|.{{PressureObserver/[[AfterPenaltyRecordMap]]}}[source] [=map/exists]:
+              <ol>
+                <li>
+                  Set |observer|.{{PressureObserver/[[AfterPenaltyRecordMap]]}}[|source|] to |record|.
+                </li>
+                <li>
+                  [=iteration/Continue=].
+                </li>
+              </ol>
             </li>
             <li>
               If running [=passes rate obfuscation test=] with |observer| and |source| returns false:
               <ol>
                 <li>
                   Set |observer|.{{PressureObserver/[[AfterPenaltyRecordMap]]}}[|source|] to |record|.
                 </li>
+                <li>
+                  Set |observer|.{{PressureObserver/[[ChangesCountMap]]}}[|source|] to 0.
+                </li>
                 <li>
                   Create timer of |observer|.{{PressureObserver/[[PenaltyDuration]]}} duration with the following callback:
                   <ol>
                     <li>
-                      Let |record| be the result of taking |observer|.{{PressureObserver/[[AfterPenaltyRecordMap]]}}[|source|].
+                      Let |record| be |observer|.{{PressureObserver/[[AfterPenaltyRecordMap]]}}[|source|].
```

Comment:
Since unobserve() and disconnect() also manipulate `[[AfterPenaltyRecordMap]]`, this entry is no longer guaranteed to exist. You need to cover this case and probably just bail out the entry is not present.